src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,113 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
-    @Override
-    public void add(T value) {
+    private static final int DEFAULT_SIZE = 10;
+    private T[] values;
+    private int size;
+    private T removedObject;
```

Comment:
```suggestion
```

Reply:
Distance between variable 'removedObject' declaration and its first usage is 4, but allowed 3.  
Consider making that variable final if you still need to store its value in advance (before method calls that might have side effects on the original value). [VariableDeclarationUsageD
istance]. This is the reason why I can not do this

Reply:
Got it, but making if a class field is not an option, here I missed that we don't need `resize()` at all (java.util.ArrayList doesn't shrink), so that will fix your checkstyle issue, in other cases make it `final` if needed